pandas/core/groupby/groupby.py

```diff
@@ -2707,6 +2707,8 @@ def resample(self, rule, *args, **kwargs):
         """
         from pandas.core.resample import get_resampler_for_grouping
 
+        self.as_index = True
```

Comment:
This changes the state of the groupby object, meaning results will change if the users performs subsequent operations. In addition, I think groupby(...).resample(...) should behave differently if `as_index` is True or False.

Reply:
HI @rhshadrach, really appreciate the feedback - the more the better please (especially any tips when it comes to working with repo are very welcomed)
I've updated the logic.


Now:
* depending on if we use as_index = False vs as_index = True, we'll get different behaviour
* the groupby's as_index won't get overwritte
* All the other cases provided in BUGs descriptions work fine (didn't include them tho, since some of the cases included were throwing FeatureWarning)


I'm not sure if this is an optimal solution, and not sure about variable name, could you provide me with some more feedback?
Thanks !

Reply:
Looks like I was wrong above, `as_index` shouldn't have an impact here. I think your prior change was in the right direction now, but as I mentioned it is changing state. In `Resampler.aggregate`, we can safely do this as:

```
with com.temp_set_attr(self._groupby, 'as_index', True):
    result = ResamplerWindowApply(self, func, args=args, kwargs=kwargs).agg()
```

Reply:
Thanks, updated